### PR TITLE
Refactor azure pipeline tasks

### LINKED ISSUE #388: Investigate whether we should update azure pipelines
### Description

We are currently using the `PublishBuildArtifacts@1` task in our Azure pipeline. According to the [Azure documentation](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-build-artifacts?view=azure-devops): this task has been deprecated. We should use [Pipeline Artifacts](https://docs.microsoft.com/en-us/azure/devops/pipelines/artifacts/pipeline-artifacts?view=azure-devops&tabs=yaml) instead.

Pipeline Artifacts is for TFS 2017 and later. We satisfy that requirement. (TFS stands for Team Foundation Server, which is the [previous name](https://azure.microsoft.com/en-us/services/devops/server/) for Devops Server; [this SO](https://stackoverflow.com/questions/60121094/how-is-the-azure-devops-service-version-determined) explains how to determine the version of TFS you have).

Also verify whether we need the `WEBSITE_RUN_FROM_PACKAGE` env var in App Service. This was not set on staging, but deploys were still working there. I set it just in case so that all the environments are consistent.

### Acceptance Criteria

- [ ] Stop using `PublishBuildArtifacts` in our Azure deploy pipeline
- [ ] Use Pipeline Artifacts instead
- [ ] Verify whether or not Pipeline Artifacts requires the `WEBSITE_RUN_FROM_PACKAGE` env var in App Service

### LINKED ISSUE #450: Release builds are including too many artifacts
### Description

Release builds are including a number of artifacts that are not required for production. We should tighten it up. May be related to #388 

### Acceptance Criteria

- [ ] Only required files should be included in a deploy

### PR DESCRIPTION
**Pull request by rocketnova**
## Ticket

Resolves #388 
Resolves #450

## Changes

- Replaces necessity of zipping symlinks by using `yarn install --no-bin-links`
- Publishes only the files needed to run `yarn start` and no extra files
- Replaces `NodeAndNpmTool@1` with `NodeTool@0` 
- Replaces `PublishBuildArtifacts@1` with `PublishPipelineArtifact@1`
- Disables Next.js telemetry

## Context

This PR makes changes to the Azure Devops pipelines build task. The are two main changes:

- This PR removes the `zip --symlink` process, which can be a [security vulnerability](https://security.stackexchange.com/questions/73718/how-zip-symlink-works). This was previously needed because `yarn install`/`npm install` will create a number of symlinks from `node_modules/.bin` to various package binaries. We can avoid needing to zip symlinks by using [`yarn install --no-bin-links`](https://classic.yarnpkg.com/en/docs/cli/install/#toc-yarn-install-no-bin-links). 
  - Note: As a result, this PR updates `package.json` to use the non-symlink binaries to ensure functionality in both local and production environments.
  - Note: We want to publish a zip file as an artifact, rather than just the files themselves, because Azure App Service provides a [deploy by zip package](https://docs.microsoft.com/en-us/azure/app-service/deploy-run-package) functionality.
- This PR only zips up and publishes the files CST needs to run `yarn start`. Previously, we were zipping the entire git repo (including `.git`), which includes way too many files. The fewer files deployed, the smaller the attack surface.

There are a few minor changes:

- Replacing [deprecated](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/publish-build-artifacts?view=azure-devops) `PublishBuildArtifacts@1` with recommended `PublishPipelineArtifact@1`
- Replacing node task (no documentation available for `NodeAndNpmTool@1`, whereas `NodeTool@0` is the [officially supported](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/tool/node-js) node task)
- Disabling [next.js telemetry](https://nextjs.org/telemetry)

## Testing

Build this branch and deploy to Internal. Test to ensure that all functionality still works as expected.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -4,47 +4,87 @@
...
+        next-i18next.config.js
+        server-preload.js
+      targetFolder: '$(Build.ArtifactStagingDirectory)/$(Build.BuildId)'
+    displayName: 'Copy files needed for yarn start'
 
+  # Zip up only the necessary files
+  # Docs: https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/utility/archive-files
   - task: ArchiveFiles@2
     inputs:
-      rootFolderOrFile: "$(build.sourcesDirectory)"
```
Comment (kalvinwang) [inline azure-pipelines.yml:32]: do you happen to know what it is about Azure Pipelines variables or something else that made these non-capitalized variables work? i see in the official docs they're capitalized and nothing about how they aren't case sensitive...

**Review by kalvinwang — APPROVED** [review]
Lgtm, super helpful detail in PR description and inline comments, thank you!

File: azure-pipelines.yml (hunk 1 above)
Comment (rocketnova) [reply azure-pipelines.yml:32]: They are definitely case insensitive. There are docs (hidden deeper) that use lower case or camel case or title case.

**Comment by rocketnova** [comment]
Note: CDT reviewed and approved this PR today.

**Pull request merged**